pkg/kubelet/cm/helpers_linux.go

```diff
@@ -119,8 +118,28 @@ func HugePageLimits(resourceList v1.ResourceList) map[int64]int64 {
 
 // ResourceConfigForPod takes the input pod and outputs the cgroup resource config.
 func ResourceConfigForPod(pod *v1.Pod, enforceCPULimits bool, cpuPeriod uint64, enforceMemoryQoS bool) *ResourceConfig {
+	inPlacePodVerticalScalingEnabled := utilfeature.DefaultFeatureGate.Enabled(kubefeatures.InPlacePodVerticalScaling)
 	// sum requests and limits.
-	reqs, limits := resource.PodRequestsAndLimits(pod)
```

Comment:
This previously ignored the InPlacePodVerticalScaling feature gate for everything except for huge-pages.  It now considers it for all resources.  I haven't convinced myself it needs to consider it at all since I *think* this is only called prior to the pod scheduling and possibly being resized.  @vinaykul , any thoughts?

Reply:
I'm not sure what is of concern, could you please provide some more context?

AFAIK, the function gathers the pod requests and limits at the time `ResourceConfigForPod` is invoked, and uses that information to decide whether to update the pod cgroup numbers before or after updating containers depending on whether the resize is a net increase or decrease respectively.

Reply:
I guess the strangeness/asymmetry here is that for cpu and memory we calculated directly from PodSpec but for hugepages we consired whether InPlacePodVerticalScaling has been enabled (and take the values from pod spec or status). I don't fully understand this

Reply:
Yes, that was it. Before this change `ResourceConfigForPod` got request/limits from `resource.PodRequestsAndLimits` which only pulls values from the requests of containers/init containers while hugepages considered InPlacePodVerticalScaling.

Reply:
Any comments @vinaykul? I think this PR looks good to be merged except I'd like to understand this particular detail